### PR TITLE
Feat: upgrade deps and expose pure esm

### PR DESCRIPTION
**Pull request by stephhuynh18**
# Upgrade deps and expose pure esm  - #511 

## Description

Closes https://github.com/ceramicnetwork/ceramic-anchor-service/issues/511

I was not able to upgrade `typeorm` to the latest version. This shall be done in a future task.

## How Has This Been Tested?

Tests can run

### CODE REVIEW

**Comment by stephhuynh18** [comment]
It failed once with the follow error:
```
Test suite failed to run

    Pubsub is not started

      at Gossipsub.unsubscribe (node_modules/libp2p-interfaces/src/pubsub/index.js:747:13)
      at Gossipsub.unsubscribe [as _unsubscribeAdapter] (node_modules/libp2p-gossipsub/src/index.js:759:15)
      at Gossipsub.unsubscribe (node_modules/libp2p/src/pubsub-adapter.js:51:14)
      at unsubscribe (node_modules/ipfs-core/cjs/src/components/pubsub.js:30:19)
          at runMicrotasks (<anonymous>)

  ● Test suite failed to run

    Pubsub is not started

      at Gossipsub.unsubscribe (node_modules/libp2p-interfaces/src/pubsub/index.js:747:13)
      at Gossipsub.unsubscribe [as _unsubscribeAdapter] (node_modules/libp2p-gossipsub/src/index.js:759:15)
      at Gossipsub.unsubscribe (node_modules/libp2p/src/pubsub-adapter.js:51:14)
      at unsubscribe (node_modules/ipfs-core/cjs/src/components/pubsub.js:30:19)
          at runMicrotasks (<anonymous>
```

**Review by stbrody — COMMENTED** [review]
lgtm if Sergey's happy with it

File: src/services/blockchain/__tests__/eth-bc-service.test.ts [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
 
-import BlockchainService from '../blockchain-service'
-import EthereumBlockchainService, { MAX_RETRIES } from '../ethereum/ethereum-blockchain-service'
+import { BlockchainService } from '../blockchain-service.js'
+import { EthereumBlockchainService, MAX_RETRIES } from '../ethereum/ethereum-blockchain-service.js'
 import { BigNumber } from 'ethers'
 import type { FeeData } from '@ethersproject/abstract-provider'
 import { ErrorCode } from '@ethersproject/logger'
-import type { TransactionRequest } from '@ethersproject/abstract-provider'
+import { JsxEmit } from 'typescript'
```
Comment (ukstv) [inline src/services/blockchain/__tests__/eth-bc-service.test.ts:17]: @stephhuynh18 Why is it needed here?

**Review by ukstv — APPROVED** [review]
mod JsxEmit import

File: src/services/blockchain/__tests__/eth-bc-service.test.ts (hunk 1 above)
Comment (stephhuynh18) [reply src/services/blockchain/__tests__/eth-bc-service.test.ts]: I have no idea =)

**Pull request merged**